src/Data/List/Lazy.purs

```diff
@@ -70,8 +70,8 @@ module Data.List.Lazy
   , groupBy
   , partition
 
-  , nub
-  , nubBy
+  , nubEq
```

Comment:
It looks like this module no longer exports `nub` or `nubBy`. I think we should wait until we can provide an Ord-based nub before renaming the existing ones to `nubEq` and `nubByEq`.

Reply:
Does this mean we also need to add a `sort` for `Data.List.Lazy` too? That might be pretty involved. Would we want to match the Haskell version (see link in first post of this PR).

Reply:
Not necessarily - I would prefer to just leave the functions in this module alone then, if adding a sort implementation for use in an Ord-based nub would be involved. I just don’t want to remove nub from this module’s exports at any point; we shouldn’t require downstream users to rewrite existing uses of nub unnecessarily.

Reply:
What about making Ord-based (according to the signature) `nub` and `nubBy`, which actually map to the slower Eq-based `nubEq` and `nubByEq` internally. We can add a comment in the docs that these versions are currently just as slow (or maybe even slightly slower) as the the Eq versions. Then later on, we can make the performance-boosting fix without affecting the API.

Reply:
I definitely don’t want to do that, it would be really misleading. I think I’d be very annoyed as a user if this function posed as O(n log n) by having an Ord signature, and then I used it and found that it was actually quadratic after it bit me in production.

Reply:
I'd be happy to dive into writing the correct Ord-based `nub` and `nubBy`, but I don't yet have a good strategy in mind for that. Could you point me to some resources to help get me going, or maybe some pseudocode for how you would go about writing this?

Reply:
I don’t know of a resource and I’m not sure exactly how I’d go about it off the top of my head, sorry. I’d be happy with just leaving this module alone for now though. It does of course mean that we can’t provide an Ord-based nub until the next round of breaking changes, but if we do find ourselves wanting to get one in before then we can always add a `nubOrd` temporarily.

Reply:
Just did some research and I think this is the ideal approach for sorting, and can be adapted for `nub`:
- Build a heap from the lazy list. This requires reading through the entire lazy list, and won't work on infinite lists.
- Lazily grab the head of the heap when consuming each element. This has a small performance boost versus traditional sorting if you only want a few elements.

But for our case, I think it's fine to forgoe the small perf boost for purely-lazy situations and just call `toFoldable`/`nub`(from list or array)/`fromFoldable`. This will still be O (n log n).

Reply:
Eh, I dunno, all of the options have downsides. What do other people think? I think my preference is
1. Leave this module alone for now, switch nub to use Ord in the next round of breaking changes
2. Implement sort and Ord-based nub for lazy lists
3. Provide a nub with Ord in the type signature which is really an Eq-based nub

Reply:
Oh to clarify, when I said “not necessarily,” that was only in response to “should I add sort,” not in response to “should I do it the way Haskell did.” I would suggest starting from Haskell’s implementation if you were going to look into implementing sort.

Reply:
Oh sorry, I only just saw your most recent comment. Going via strict lists or arrays sounds fine to me (I don’t have a preference for which).

Reply:
Ah, hm, actually maybe not. I think having sorting work by going via strict lists or arrays would be okay, but a nub function on lazy lists really ought to be able to handle eg `take 1 $ nub $ repeat “hi”`, I think. Especially if the current one does handle that (I think it does).

Reply:
That's a good point, since this currently works:
```purs
take 2 $ nub $ cycle $ range 1 3
```